src/js/wrap.cpp

```diff
@@ -90,17 +115,32 @@ namespace ccf::js
     }
     size_t stack_size = 1024 * 1024;
     size_t heap_size = 100 * 1024 * 1024;
+    std::chrono::seconds default_max_execution_time{240};
+
     const auto jsengine = tx->ro<ccf::JSEngine>(ccf::Tables::JSENGINE);
     const std::optional<JSRuntimeOptions> js_runtime_options = jsengine->get();
 
     if (js_runtime_options.has_value())
     {
       heap_size = js_runtime_options.value().max_heap_bytes;
       stack_size = js_runtime_options.value().max_stack_bytes;
+      default_max_execution_time =
+        std::chrono::seconds{js_runtime_options.value().max_execution_time};
     }
 
     JS_SetMaxStackSize(rt, stack_size);
     JS_SetMemoryLimit(rt, heap_size);
+    UntrustedHostTime* host_time = new UntrustedHostTime();
+    const auto curr_time = ccf::get_enclave_time();
+    host_time->start_time = curr_time;
+    host_time->max_execution_time = default_max_execution_time;
+    JS_SetInterruptHandler(rt, js_custom_interrupt_handler, host_time);
+  }
+
+  Runtime::~Runtime()
+  {
+    JS_FreeRuntime(rt);
+    JS_SetInterruptHandler(rt, NULL, NULL);
```

Comment:
```suggestion
    JS_SetInterruptHandler(rt, NULL, NULL);
    JS_FreeRuntime(rt);
```

I don't think that resetting the interrupt handler should be necessary. But if we're going to, we must do that _before_ freeing the runtime.

Reply:
Yes for the latter! As for not resetting the interrupt handler, it does look like it shouldn't have an impact. But given that it will set the (now) deleted raw memory (`host_time`) to `NULL` (preventing future double deletes, though it's unlikely to happen in this case) it seems fine to keep it.

Reply:
+1 for the cautious reset, it doesn't hurt.